### PR TITLE
Migrate  clone and gettid to Rust

### PR DESCRIPTION
**Pull request by sporksmith**

### CODE REVIEW

**Review by stevenengler — APPROVED** [review]

File: src/main/host/context.rs [hunk 1]
```diff
@@ -131,9 +131,16 @@ impl<'a> ThreadContextObjs<'a> {
...
+            .thread_borrow(self.tid)
+            .unwrap()
+            .clone(self.host.root());
+        let res = {
+            let thread = threadrc.borrow(self.host.root());
+            let mut ctx = ThreadContext::new(self.host, &process, &thread);
+            f(&mut ctx)
+        };
+        threadrc.safely_drop(self.host.root());
+        res
```
[the remark this answers is not in the document]
Comment (sporksmith) [reply src/main/host/context.rs:146]: Done

File: src/main/host/syscall/handler/clone.rs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        /* ctid */*const libc::pid_t,
+        /* newtls */*const libc::c_void)]
+    pub fn clone(
+        ctx: &mut SyscallContext,
+        flags_and_exit_signal: i32,
+        child_stack: ForeignPtr<()>,
+        ptid: ForeignPtr<libc::pid_t>,
+        ctid: ForeignPtr<libc::pid_t>,
+        newtls: u64,
+    ) -> SyscallResult {
```
[the remark this answers is not in the document]
Comment (sporksmith) [reply src/main/host/syscall/handler/clone.rs]: done

File: src/main/host/syscall/handler/clone.rs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            let childrc = ctx.objs.process.thread_borrow(child_tid).unwrap();
+            let child = childrc.borrow(ctx.objs.host.root());
+            child.set_tid_address(ctid);
+        }
+
+        Ok(libc::pid_t::from(child_tid).into())
+    }
+
+    #[log_syscall(/* rv */libc::pid_t)]
+    pub fn gettid(ctx: &mut SyscallContext) -> SyscallResult {
```
[the remark this answers is not in the document]
Comment (sporksmith) [reply src/main/host/syscall/handler/clone.rs]: done

File: src/main/host/syscall/handler/clone.rs [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    pub fn clone(
+        ctx: &mut SyscallContext,
+        flags_and_exit_signal: i32,
+        child_stack: ForeignPtr<()>,
+        ptid: ForeignPtr<libc::pid_t>,
+        ctid: ForeignPtr<libc::pid_t>,
+        newtls: u64,
+    ) -> SyscallResult {
+        let raw_flags = flags_and_exit_signal & !0xff;
+        let raw_exit_signal = flags_and_exit_signal & 0xff;
```
[the remark this answers is not in the document]
Comment (sporksmith) [reply src/main/host/syscall/handler/clone.rs]: done

**Pull request merged**